### DISCUSSION TITLE
Why these error are happening in Vite-react project?

### DISCUSSION THREAD

**Discussion by sachinshilwal** [TanStack/table#4083] [Q&A]
When I am importing react table like this:
`import { useTable } from 'react-table/src';`

I get error like this in my console:
![image](https://user-images.githubusercontent.com/34100453/176993583-add8cec2-646f-4fd5-a1a9-b3cfdc356c93.png)

But when I import like this:
`import { useTable} from 'react-table';`
I get error like this in my browser console
![image](https://user-images.githubusercontent.com/34100453/176993651-687a05e9-94d5-4bec-820a-04cb788aeaaa.png)

My project is vite. Any help would be appreciated.

**Comment by nrayburn-tech** [comment]
Vite expects jsx to only be in files that have the .jsx or .tsx extension, otherwise it doesn't process the jsx.  So, `import { useTable } from 'react-table/src';` is likely to not work (and I am not sure why you would be using that here anyways.)

I can't speak for the other error, you would likely need to provide more information around how you are using `column.getHeaderGroupProps` for anybody to be able to help you.

**Comment by j0j032** [comment]
Hey, I get exectly the same issue.
so I went there `node_modules/react-table/src/publicUtils.js`
and I went through the error by replacing the extension of `publicUtils.js` for `publicUtils.jsx `and relaunch `pnpm run dev`
I'm not sure is a good thing to edit filename in node_modules  but it is not a big deal and it works.

So the error looks like vite doesn't want to compile if a file has `.js` extension and contains `.jsx` syntax .
